### PR TITLE
markets file, fixes

### PR DESCRIPTION
**Pull request by turuslan**
- fd usage metric.
- hasher returns outcome.
- remove `ipld::Verifier` class.
- `unixfs::wrapFile` from file reader (not from memory).
- memory indexed car, with append.
- set max fd limit for node.
- fix sync same parent weight.
- fix sync branch not found null.

### CODE REVIEW

File: core/common/bytes_stream.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+#pragma once
+
+#include <boost/iostreams/device/array.hpp>
+#include <boost/iostreams/stream.hpp>
+
+#include "common/span.hpp"
+
+namespace fc {
+  auto bytesIstream(BytesIn bytes) {
```
Comment (Alexey-N-Chernyshov) [inline core/common/bytes_stream.hpp:14]: `bytesToIostream` more meaningful in my opinion.

File: core/common/fd_usage.hpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cstdlib>
+
+namespace fc {
+  inline size_t fdUsage();
```
Comment (Alexey-N-Chernyshov) [inline core/common/fd_usage.hpp:11]: Please, add description like `Returns the number of fds opened by the current process.`

File: core/common/fd_usage.hpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  return r / sizeof(proc_fdinfo);
+}
+
+#elif __linux__
+
+#include <filesystem>
+
+size_t fc::fdUsage() {
+  size_t n{};
+  std::filesystem::directory_iterator it{"/proc/self/fd"}, end;
```
Comment (Alexey-N-Chernyshov) [inline core/common/fd_usage.hpp:40]: `std::distance(std::filesystem::directory_iterator("/proc/self/fd"), std::filesystem::directory_iterator::end());`

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: core/storage/ipld/memory_indexed_car.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    OUTCOME_TRY(header, codec::cbor::decode<storage::car::CarHeader>(item));
+    ipld->roots = std::move(header.roots);
+    while (offset < car_size) {
+      varint = codec::uvarint::readBytes(ipld->reader, item);
+      // incomplete item or zero padding
+      if (varint == 0 || item.size() == 0) {
+        break;
+      }
+      offset += varint + item.size();
+      BytesIn input{item};
```
Comment (wer1st) [inline core/storage/ipld/memory_indexed_car.cpp:46]:
```suggestion
      const BytesIn input{item};
```

**Review by wer1st — APPROVED** [review]

File: core/storage/ipld/memory_indexed_car.cpp (hunk 4 above)
Comment (turuslan) [reply core/storage/ipld/memory_indexed_car.cpp:46]: No, `CID::read` modifies it

Comment (wer1st) [reply core/storage/ipld/memory_indexed_car.cpp:46]: Ok

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**